htdocs/filefunc.inc.php

```diff
@@ -191,7 +192,16 @@
 }
 //print "found=".$found." dolibarr_main_url_root=".$dolibarr_main_url_root."\n";
 if (! $found) $tmp=$dolibarr_main_url_root; // If autodetect fails (Ie: when using apache alias that point outside default DOCUMENT_ROOT).
-else $tmp='http'.(((empty($_SERVER["HTTPS"]) || $_SERVER["HTTPS"] != 'on') && (empty($_SERVER["SERVER_PORT"])||$_SERVER["SERVER_PORT"]!=443))?'':'s').'://'.$_SERVER["SERVER_NAME"].((empty($_SERVER["SERVER_PORT"])||$_SERVER["SERVER_PORT"]==80||$_SERVER["SERVER_PORT"]==443)?'':':'.$_SERVER["SERVER_PORT"]).($tmp3?(preg_match('/^\//',$tmp3)?'':'/').$tmp3:'');
+else {
+	$uri = 'http'.($_SERVER['HTTPS'] == 'on' ? 's' : '');
```

Comment:
I experienced this with some version of IIS of exotic web server, but i can't remember which one. It also depends if you use a proxy or not. I know this part of code is very critical because it takes a long time to have this working on most OS/web server/versions/proxy/setup.

I can't reproduce the bad detection on an https installation. Which web server/version do you use ? In your case, does server run on port 443 ? Do you use a proxy ?